### PR TITLE
Add autocompletion for password reset

### LINKED ISSUE #27885: Changing the password does not work properly on macOS/Safari
In particular, Safari enters a new, strong password in the old password field. This makes it impossible to change to a new, automatically generated strong password that is immediately kept safe in the macOS keychain.

Generally, websites use three fields, the first one the old password and the next two the new passwords (which must be identical). I urge you to adopt that pattern as well, or instrument the password fields so Safari can work its password convenience magic properly.

### PR DESCRIPTION
**Pull request by CarlSchwan**
Using autocomplete="current-password" and autocomplete="new-password"
will help browser with integrated password managers to generate safe
password for the users.

See https://developer.apple.com/documentation/security/password_autofill/enabling_password_autofill_on_an_html_input_element
and https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Input/password#allowing_autocomplete.

Close #27885

Signed-off-by: Carl Schwan <carl@carlschwan.eu>

### CODE REVIEW

**Comment by eneiluj** [comment]
@szaimen Are you sure you want to assign this PR? Maybe you wanted to request reviews? :grin:

**Comment by szaimen** [comment]
> Are you sure you want to assign this PR? Maybe you wanted to request reviews? 😁

You are right. I misclicked 😅

**Review by julien-nc — COMMENTED** [review]
Nice! This could also be done in login page and password reset page.
`core/src/components/login/LoginForm.vue`
`core/src/components/login/UpdatePassword.vue`

**Comment by CarlSchwan** [comment]
> Nice! This could also be done in login page and password reset page.
> `core/src/components/login/LoginForm.vue`
> `core/src/components/login/UpdatePassword.vue`

Good idea :) Just did that and also unified the `autocapitalize="none" autocorrect="off"` behavior with the two places you linked. Not sure it's really needed since I expect browsers to be clever enough to not autocorrect password fields by default but at least it's consistent :)

**Comment by szaimen** [comment]
Seems like the compile bot is broken today :(

File: core/src/components/login/LoginForm.vue [hunk 1]
```diff
@@ -60,6 +60,7 @@
 					type="text"
 					name="user"
 					autocapitalize="off"
```
Comment (Pytal) [inline core/src/components/login/LoginForm.vue:62]:
```suggestion
					autocapitalize="none"
```
For consitency :)

File: core/src/components/login/UpdatePassword.vue [hunk 2]
```diff
@@ -29,6 +29,9 @@
 					v-model="password"
 					type="password"
 					name="password"
+					autocomplete="current-password"
```
Comment (eneiluj) [inline core/src/components/login/UpdatePassword.vue:32]:
```suggestion
					autocomplete="new-password"
```
as it's the place where one sets a new password :grin:

**Review by julien-nc — CHANGES_REQUESTED** [review]

**Review by julien-nc — APPROVED** [review]

**Comment by szaimen** [comment]
Solved the conflicting files. But I fear there is still the need to compile and push the js.
`/compile amend /` unfortunatlely didn't work the last times...

**Review by Pytal — APPROVED** [review]

**Comment by artonge** [comment]
I compiled the files, but can't push on @CarlSchwan's branch :(

**Comment by CarlSchwan** [comment]
I will compile the files and push them after lunch

**Comment by szaimen** [comment]
Unfortunately the fork seems to be still unpushable...

**Pull request merged**